kernel/univ.ml

```diff
@@ -944,30 +944,33 @@ let eq_puniverses f (x, u) (y, u') =
 
 module UContext =
 struct
-  type t = Instance.t constrained
+  type t = Names.Name.t array * Instance.t constrained
 
-  let make x = x
+  let make names x = (names, x)
```

Comment:
Maybe assert that length names = length univs here, then we can simplify a bit is_empty

Reply:
OK, done, and this answers also @ppedrot's remark.